lib/resources/xml.rb

```diff
@@ -20,7 +20,18 @@ def parse(content)
     end
 
     def value(key)
-      REXML::XPath.each(@params, key.first.to_s).map(&:text)
+      output = []
+      REXML::XPath.each(@params, key.first.to_s) do |element|
+        if element.is_a?(REXML::Attribute)
+          output.push(element.to_s)
+        elsif element.is_a?(REXML::Element)
+          output.push(element.text)
+        else
+          raise Inspec::Exceptions::ResourceSkipped, 'We have encountered an unknown class from REXML'
```

Comment:
Let's fail this, and let's give a little more help to the user:

```ruby
raise Inspec::Exceptions::ResourceFailed, "Unknown XML object received (#{element.class}): #{element}"
```

That way the user can either try to fix on their own, or they'll be able to give us more helpful debugging information to fix a bug.